server/container_create_linux.go

```diff
@@ -816,6 +816,16 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 		makeOCIConfigurationRootless(specgen)
 	}
 
+	if err := s.nri.createContainer(ctx, specgen, sb, ociContainer); err != nil {
+		return nil, err
```

Comment:
I guess I don't understand why for some nri calls (such as postCreateContainer) you choose to warn, but for some others (like this one) you choose to error out.

Reply:
So the general theme is that
- errors in NRI functions which could have resulted in changes to containers cause the triggering CRI call to error out
- errors in calls to NRI functions which can't alter container state are logged (with a warning or error), but do not error out from the triggering CRI call

Potentially state-altering calls are create, update and stop. Because stop is semantically a post-stop, there we log errors but don't cause the CRI call to error out.